config/sections/pages.php

```diff
@@ -315,12 +318,49 @@
 			return $blueprints;
 		},
 	],
+	// @codeCoverageIgnoreStart
+	'api' => function () {
+		return [
+			[
+				'pattern' => 'delete',
+				'method'  => 'DELETE',
+				'action'  => function () {
+					$section = $this->section();
+
+					// check if batch deletion is allowed
+					if ($section->batch() === false) {
+						throw new PermissionException(
+							message: 'The section does not support batch actions'
+						);
+					}
+
+					$ids = $this->requestBody('ids');
+					$min = $section->min();
+
+					// check if the section has enough pages after the deletion
+					if ($section->total() - count($ids) < $min) {
+						throw new Exception(
+							message: I18n::template('error.section.pages.min.' . I18n::form($min), [
+								'min'     => $min,
+								'section' => $section->headline()
+							])
+						);
+					}
+
+					$section->models()->delete($ids);
+					return true;
```

Comment:
`Todo:` This code is almost the same as in the files section. I think we should DRY this with a method included in the `batch` mixin.

Reply:
I've moved it all into the batch mixin. Works great and fits very well.